mage/algorithms/machine-learning-graph-analytics/kmeans-clustering-algorithm.md

```diff
@@ -0,0 +1,37 @@
+---
+id: k-means-clustering-algorithm
+title: K-means Clustering
+sidebar_label: K-means Clustering
+---
+
+## Description
+
+The k-means algorithm clusters given data by trying to separate samples in `n` groups of equal variance by minimizing a criterion known as 
+within-the-cluster sum-of-squares. 
+Algorithm requires the number of clusters to be specified beforehand. This means it is not possible for algorithm to determine
+what is the best number of clusters to use, but it is up to user to find it. It scales well to large numbers of samples and has been used 
+across a large range of application areas in many different fields such as recommendation engine, fraud detection and so on.
+
+
+![k-means clustering](../../data/algorithms/machine-learning-graph-analytics/kmeans-clustering.png)
+> K-means clustering run on top of Node2Vec algorithm. Numbers represent cluster.
```

Comment:
```suggestion
> K-means clustering run on node2vec output; clusters are numbered.
```

Reply:
BTW, is the node2vec part important here? I’ve looked at the above image and don’t see how it relates to node2vec

Reply:
It is not important, I will remove it